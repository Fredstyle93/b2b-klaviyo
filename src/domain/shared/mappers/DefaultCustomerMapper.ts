import { Customer } from '@commercetools/platform-sdk';
import { CustomerMapper } from './CustomerMapper';
import {
    getCTCustomerAddressForKlaviyo,
    getPhoneNumber,
} from '../../eventSync/eventProcessors/customer/utils/CustomerAddressUtils';
import { mapCTAddressToKlaviyoLocation } from '../../eventSync/eventProcessors/customer/mappers/CTAddressToKlaviyoLocationMapper';

export class DefaultCustomerMapper implements CustomerMapper {
    public mapCtCustomerToKlaviyoProfile(customer: Customer): ProfileRequest {
        const address = getCTCustomerAddressForKlaviyo(customer);
        const {
            email,
            firstName: first_name,
            lastName: last_name,
            title,
            companyName: organization,
            id: external_id,
        } = customer;

        return {
            data: {
                type: 'profile',
                attributes: {
                    email,
                    external_id,
                    first_name,
                    last_name,
                    title,
                    phone_number: getPhoneNumber(address),
                    organization,
                    location: mapCTAddressToKlaviyoLocation(address),
                },
            },
        };
    }
}
