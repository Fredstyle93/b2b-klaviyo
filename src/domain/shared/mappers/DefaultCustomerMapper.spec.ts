import { DefaultCustomerMapper } from './DefaultCustomerMapper';
import { sampleCustomerApiResponse } from '../../../test/testData/ctCustomerApi'

const customerMapper = new DefaultCustomerMapper();
describe('map CT customer to Klaviyo profile', () => {
    it('should map a commercetools customer to a klaviyo profile', () => {
        const klaviyoEvent = customerMapper.mapCtCustomerToKlaviyoProfile(sampleCustomerApiResponse);
        expect(klaviyoEvent).toMatchSnapshot();
    });
});
