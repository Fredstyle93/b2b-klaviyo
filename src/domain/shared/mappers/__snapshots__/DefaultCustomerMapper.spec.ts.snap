// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map CT customer to Klaviyo profile should map a commercetools customer to a klaviyo profile 1`] = `
{
  "data": {
    "attributes": {
      "email": "Alfred_Lebsack@hotmail.com",
      "external_id": "77d8e6f4-2dee-4ac9-bd21-4035b6ba76d8",
      "first_name": "Jailyn",
      "last_name": "Hilll",
      "location": {
        "address1": "12, First Street",
        "address2": "",
        "city": "Brookefort",
        "country": "NL",
        "region": undefined,
        "zip": "12345",
      },
      "organization": undefined,
      "phone_number": "+393234567892",
      "title": "Miss",
    },
    "type": "profile",
  },
}
`;
